components/server/src/feature-flag/featureflag-controller.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { injectable, inject } from "inversify";
+import * as express from "express";
+import { Config } from "../config";
+import { getExperimentsClientForBackend } from "@gitpod/gitpod-protocol/lib/experiments/configcat-server";
+import { User } from "@gitpod/gitpod-protocol";
+
+@injectable()
+export class FeatureFlagController {
+    @inject(Config) protected readonly config: Config;
+
+    get apiRouter(): express.Router {
+        const router = express.Router();
+        this.addSlowDatabaseFeatureFlagHandler(router);
+        return router;
+    }
+
+    protected addSlowDatabaseFeatureFlagHandler(router: express.Router) {
+        router.get("/slow-database", async (req, res) => {
+            if (!User.is(req.user)) {
+                res.sendStatus(400);
```

Comment:
q: why not 401 for unauthenticated? if there is a reason, please add a line of doc in here.

Reply:
I took this from the `code-sync-service` which returns `400` in this case, see eg:

https://github.com/gitpod-io/gitpod/blob/9072a7f945a23c613d0197df7baa89c74c818276/components/server/src/code-sync/code-sync-service.ts#L288-L291

and various other places. `401` is a better code though and it's also used in various places too, so I've changed to use `401`.